polaris-react/src/components/Badge/Badge.tsx

```diff
@@ -124,7 +136,12 @@ export function Badge({
   return (
     <span className={className}>
       {accessibilityMarkup}
-      {children}
+      {icon && (
+        <span className={styles.Icon}>
+          <Icon source={icon} />
```

Comment:
Thoughts on this one @chloerice? Do we really use empty `Badge`s anywhere? I think it might be the right time to make the `children` prop required if not.

Reply:
Hmmm yeah that does seem like an oversight 🤔 👀

Reply:
Should I just merge? I’m a bit worried that by introducing the icon we’re allowing consumers to create unaccessible badges 😅